resource_multi_week_calendar/models/resource_calendar.py

```diff
@@ -0,0 +1,248 @@
+        if day is None:
+            day = fields.Date.today()
+        if isinstance(day, datetime):
+            day = day.date()
+        family_size = len(self.family_calendar_ids)
+        weeks_since_epoch = math.floor(
+            (day - self._get_first_day_of_epoch_week()).days / 7
+        )
+        return (weeks_since_epoch % family_size) + 1
+
+    def _get_calendar(self, day=None):
+        self.ensure_one()
+        if not self.is_multi_week:
+            return self
+        week_number = self._get_week_number(day=day)
+        return self.family_calendar_ids.filtered(
+            lambda item: item.week_number == week_number
+        )
+
+    @api.depends(
+        "multi_week_epoch_date",
+        "week_number",
+        "family_calendar_ids",
+    )
+    def _compute_current_week(self):
+        for calendar in self:
+            current_week_number = calendar._get_week_number()
+            calendar.current_week_number = current_week_number
+            calendar.current_calendar_id = calendar._get_calendar()
+
+    @api.constrains("parent_calendar_id", "child_calendar_ids")
+    def _check_child_is_not_parent(self):
+        err_str = _(
+            "Working Time '%(name)s' may not be the Main Working Time of"
+            " another Working Time ('%(child)s') while it has a Main Working"
+            " Time itself ('%(parent)s')"
+        )
+        for calendar in self:
+            if calendar.parent_calendar_id and calendar.child_calendar_ids:
+                raise ValidationError(
+                    err_str
+                    % {
+                        "name": calendar.name,
+                        "child": calendar.child_calendar_ids[0].name,
+                        "parent": calendar.parent_calendar_id.name,
+                    }
+                )
+            # This constraint isn't triggered on calendars which have children
+            # added to them. Therefore, we also check whether our parent already
+            # has a parent.
+            if (
+                calendar.parent_calendar_id
+                and calendar.parent_calendar_id.parent_calendar_id
+            ):
+                raise ValidationError(
+                    err_str
+                    % {
+                        "name": calendar.parent_calendar_id.name,
+                        "child": calendar.name,
+                        "parent": calendar.parent_calendar_id.parent_calendar_id.name,
+                    }
+                )
+
+    def get_multi_week_epoch_date(self):
+        self.ensure_one()
+        if self.parent_calendar_id:
+            return self.parent_calendar_id.multi_week_epoch_date
+        return self.multi_week_epoch_date
+
+    @api.model
+    def _split_into_weeks(self, start_dt, end_dt):
+        # TODO: This method splits weeks on the timezone of start_dt. Maybe it
+        # should split weeks on the timezone of the calendar. It is not
+        # immediately clear to me how to implement that.
```

Comment:
This code has a possible solution: https://github.com/coopiteasy/addons/blob/4ca25c1305df5aa0c60a771f58896bccfb32e349/resource_work_time_from_contracts/models/resource_mixin.py#L43